Native Image build fails after upgrade to Micronaut 3.8.0

### Expected Behavior

The Docker image should be built w/o any error. The same Docker image was build successfully using Micronaut 3.7.5

### Actual Behaviour

See problem description below.

### Steps To Reproduce

given the following generated `DockerfileNative` 

```Dockerfile
FROM ghcr.io/graalvm/native-image:ol7-java17-22.2.0 AS graalvm
WORKDIR /home/app
COPY layers/libs /home/app/libs
COPY layers/classes /home/app/classes
COPY layers/resources /home/app/resources
COPY layers/application.jar /home/app/application.jar
RUN mkdir /home/app/config-dirs
COPY config-dirs/generateResourcesConfigFile /home/app/config-dirs/generateResourcesConfigFile
RUN native-image -cp /home/app/libs/*.jar:/home/app/resources:/home/app/application.jar --no-fallback -H:Name=application -J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.core.configure=ALL-UNNAMED -J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.core.jdk=ALL-UNNAMED -J--add-exports=org.graalvm.nativeimage.builder/com.oracle.svm.core.jni=ALL-UNNAMED -J--add-exports=org.graalvm.sdk/org.graalvm.nativeimage.impl=ALL-UNNAMED -H:ConfigurationFileDirectories=/home/app/config-dirs/generateResourcesConfigFile -H:Class=ch.onstructive.salessystem.Application
FROM frolvlad/alpine-glibc:alpine-3.12
RUN apk --no-cache update && apk add libstdc++
EXPOSE 8080
HEALTHCHECK CMD curl -s localhost:8080/endpoints/health | grep '"status":"UP"'
COPY --from=graalvm /home/app/application /app/application
ENTRYPOINT ["/app/application"]
```

The build of the Docker image fails with the following error regarding Micrometer.

```
micronaut-projects/micronaut-core#17 8.263 [1/7] Initializing...                                                                                    (7.3s @ 0.52GB)
micronaut-projects/micronaut-core#17 8.263  Version info: 'GraalVM 22.2.0 Java 17 CE'
micronaut-projects/micronaut-core#17 8.263  Java version info: '17.0.4+8-jvmci-22.2-b06'
micronaut-projects/micronaut-core#17 8.263  C compiler: gcc (redhat, aarch64, 4.8.5)
micronaut-projects/micronaut-core#17 8.263  Garbage collector: Serial GC
micronaut-projects/micronaut-core#17 8.264  8 user-specific feature(s)
micronaut-projects/micronaut-core#17 8.264  - ch.onstructive.salessystem.graalvm.SalesServerHibernateFeature
micronaut-projects/micronaut-core#17 8.264  - io.micronaut.buffer.netty.NettyFeature
micronaut-projects/micronaut-core#17 8.264  - io.micronaut.configuration.hibernate.jpa.graal.HibernateFeature
micronaut-projects/micronaut-core#17 8.264  - io.micronaut.core.graal.ServiceLoaderFeature
micronaut-projects/micronaut-core#17 8.264  - io.micronaut.flyway.graalvm.FlywayFeature
micronaut-projects/micronaut-core#17 8.264  - io.micronaut.http.netty.graal.HttpNettyFeature
micronaut-projects/micronaut-core#17 8.264  - io.micronaut.jackson.JacksonDatabindFeature
micronaut-projects/micronaut-core#17 8.264  - io.micronaut.jdbc.nativeimage.JdbcFeature
micronaut-projects/micronaut-core#17 32.43 [2/7] Performing analysis...  [*]                                                                       (23.9s @ 2.34GB)
micronaut-projects/micronaut-core#17 32.43   12,973 (88.23%) of 14,703 classes reachable
micronaut-projects/micronaut-core#17 32.44   14,253 (52.78%) of 27,005 fields reachable
micronaut-projects/micronaut-core#17 32.47   52,532 (57.74%) of 90,982 methods reachable
micronaut-projects/micronaut-core#17 32.47    1,361 classes,   776 fields, and 4,464 methods registered for reflection
micronaut-projects/micronaut-core#17 32.47 
micronaut-projects/micronaut-core#17 32.47 Fatal error: com.oracle.graal.pointsto.util.AnalysisError$ParsingError: Error encountered while parsing io.micronaut.configuration.metrics.binder.executor.ExecutorServiceMetricsBinder.onCreated(io.micronaut.context.event.BeanCreatedEvent) 
micronaut-projects/micronaut-core#17 32.47 Parsing context:
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.configuration.metrics.binder.executor.ExecutorServiceMetricsBinder.onCreated(ExecutorServiceMetricsBinder.java:64)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.configuration.metrics.binder.executor.ExecutorServiceMetricsBinder.onCreated(ExecutorServiceMetricsBinder.java:48)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.triggerBeanCreatedEventListener(DefaultBeanContext.java:2407)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.postBeanCreated(DefaultBeanContext.java:2385)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2309)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2251)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3016)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2906)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2853)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.loadListeners(DefaultBeanContext.java:1872)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.triggerPreDestroyListeners(DefaultBeanContext.java:1184)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.destroyBean(DefaultBeanContext.java:1144)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.destroyBean(DefaultBeanContext.java:1119)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultBeanContext.stop(DefaultBeanContext.java:406)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultApplicationContext.stop(DefaultApplicationContext.java:200)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.DefaultApplicationContext.stop(DefaultApplicationContext.java:63)
micronaut-projects/micronaut-core#17 32.47    at io.micronaut.context.LifeCycle.close(LifeCycle.java:61)
micronaut-projects/micronaut-core#17 32.47    at java.io.FileDescriptor.closeAll(FileDescriptor.java:345)
micronaut-projects/micronaut-core#17 32.47    at java.io.FileInputStream.close(FileInputStream.java:457)
micronaut-projects/micronaut-core#17 32.47    at sun.nio.cs.StreamDecoder.implClose(StreamDecoder.java:366)
micronaut-projects/micronaut-core#17 32.47    at sun.nio.cs.StreamDecoder.close(StreamDecoder.java:204)
micronaut-projects/micronaut-core#17 32.47    at java.io.InputStreamReader.close(InputStreamReader.java:192)
micronaut-projects/micronaut-core#17 32.47    at java.io.BufferedReader.close(BufferedReader.java:525)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.containers.CgroupUtil.readStringValue(CgroupUtil.java:61)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.containers.CgroupSubsystemController.getStringValue(CgroupSubsystemController.java:66)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.containers.CgroupSubsystemController.getLongValue(CgroupSubsystemController.java:125)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.containers.cgroupv2.CgroupV2Subsystem.getLongVal(CgroupV2Subsystem.java:56)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.containers.cgroupv2.CgroupV2Subsystem.getCpuShares(CgroupV2Subsystem.java:174)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.ContainerInfo.getCpuShares(ContainerInfo.java:49)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.Containers.activeProcessorCount(Containers.java:129)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.jdk.Target_java_lang_Runtime.availableProcessors(JavaLangSubstitutions.java:245)
micronaut-projects/micronaut-core#17 32.47    at java.util.concurrent.ForkJoinPool.<init>(ForkJoinPool.java:2564)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.jdk.ForkJoinPoolCommonAccessor.initializeCommonPool(RecomputedFields.java:367)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.jdk.ForkJoinPoolCommonAccessor.get(RecomputedFields.java:359)
micronaut-projects/micronaut-core#17 32.47    at java.util.concurrent.ForkJoinTask.fork(ForkJoinTask.java:652)
micronaut-projects/micronaut-core#17 32.47    at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:306)
micronaut-projects/micronaut-core#17 32.47    at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
micronaut-projects/micronaut-core#17 32.47    at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
micronaut-projects/micronaut-core#17 32.47    at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:686)
micronaut-projects/micronaut-core#17 32.47    at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
micronaut-projects/micronaut-core#17 32.47    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
micronaut-projects/micronaut-core#17 32.47    at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
micronaut-projects/micronaut-core#17 32.47    at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682)
micronaut-projects/micronaut-core#17 32.47    at java.lang.reflect.Executable.sharedToString(Executable.java:122)
micronaut-projects/micronaut-core#17 32.47    at java.lang.reflect.Method.toString(Method.java:412)
micronaut-projects/micronaut-core#17 32.47    at java.lang.String.valueOf(String.java:4218)
micronaut-projects/micronaut-core#17 32.47    at java.lang.AssertionError.<init>(AssertionError.java:75)
micronaut-projects/micronaut-core#17 32.47    at com.oracle.svm.core.snippets.ImplicitExceptions.throwNewAssertionErrorObject(ImplicitExceptions.java:319)
micronaut-projects/micronaut-core#17 32.47 
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.AnalysisError.parsingError(AnalysisError.java:152)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:104)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.ensureFlowsGraphCreated(MethodTypeFlow.java:83)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.getOrCreateMethodFlowsGraph(MethodTypeFlow.java:65)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.typestate.DefaultVirtualInvokeTypeFlow.onObservedUpdate(DefaultVirtualInvokeTypeFlow.java:109)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.TypeFlow.update(TypeFlow.java:558)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis$1.run(PointsToAnalysis.java:635)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.executeCommand(CompletionExecutor.java:193)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.lambda$executeService$0(CompletionExecutor.java:177)
micronaut-projects/micronaut-core#17 32.47       at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1395)
micronaut-projects/micronaut-core#17 32.47       at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
micronaut-projects/micronaut-core#17 32.47       at java.base/java.util.concurrent.ForkJoinPool.externalHelpQuiescePool(ForkJoinPool.java:2104)
micronaut-projects/micronaut-core#17 32.47       at java.base/java.util.concurrent.ForkJoinPool.awaitQuiescence(ForkJoinPool.java:3321)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.util.CompletionExecutor.complete(CompletionExecutor.java:243)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.doTypeflow(PointsToAnalysis.java:692)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.finish(PointsToAnalysis.java:680)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.PointsToAnalysis.runAnalysis(PointsToAnalysis.java:736)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.runPointsToAnalysis(NativeImageGenerator.java:731)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.doRun(NativeImageGenerator.java:564)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGenerator.run(NativeImageGenerator.java:521)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.buildImage(NativeImageGeneratorRunner.java:407)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.build(NativeImageGeneratorRunner.java:585)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.NativeImageGeneratorRunner.main(NativeImageGeneratorRunner.java:128)
micronaut-projects/micronaut-core#17 32.47 Caused by: org.graalvm.compiler.java.BytecodeParser$BytecodeParserError: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of io.micrometer.common.util.internal.logging.LocationAwareSlf4JLogger are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=io.micrometer.common.util.internal.logging.LocationAwareSlf4JLogger.
micronaut-projects/micronaut-core#17 32.47       at parsing io.micrometer.core.instrument.binder.jvm.ExecutorServiceMetrics.bindTo(ExecutorServiceMetrics.java:304)
micronaut-projects/micronaut-core#17 32.47       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.throwParserError(BytecodeParser.java:2506)
micronaut-projects/micronaut-core#17 32.47       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase$SharedBytecodeParser.throwParserError(SharedGraphBuilderPhase.java:105)
micronaut-projects/micronaut-core#17 32.47       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3367)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.handleBytecodeBlock(BytecodeParser.java:3319)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBlock(BytecodeParser.java:3164)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.build(BytecodeParser.java:1138)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.buildRootMethod(BytecodeParser.java:1030)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.GraphBuilderPhase$Instance.run(GraphBuilderPhase.java:84)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.SharedGraphBuilderPhase.run(SharedGraphBuilderPhase.java:79)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.run(Phase.java:49)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.phases.BasePhase.apply(BasePhase.java:261)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:42)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.phases.Phase.apply(Phase.java:38)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.AnalysisParsedGraph.parseBytecode(AnalysisParsedGraph.java:135)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisMethod.ensureGraphParsed(AnalysisMethod.java:685)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.lookupEncodedGraph(InlineBeforeAnalysis.java:180)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.doInline(PEGraphDecoder.java:1162)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.tryInline(PEGraphDecoder.java:1145)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.trySimplifyInvoke(PEGraphDecoder.java:1003)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.handleInvoke(PEGraphDecoder.java:957)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.GraphDecoder.processNextNode(GraphDecoder.java:817)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysisGraphDecoder.processNextNode(InlineBeforeAnalysis.java:240)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.GraphDecoder.decode(GraphDecoder.java:548)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.replacements.PEGraphDecoder.decode(PEGraphDecoder.java:833)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.phases.InlineBeforeAnalysis.decodeGraph(InlineBeforeAnalysis.java:98)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.parse(MethodTypeFlowBuilder.java:176)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlowBuilder.apply(MethodTypeFlowBuilder.java:343)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.flow.MethodTypeFlow.createFlowsGraph(MethodTypeFlow.java:93)
micronaut-projects/micronaut-core#17 32.48       ... 21 more
micronaut-projects/micronaut-core#17 32.48 Caused by: com.oracle.graal.pointsto.constraints.UnsupportedFeatureException: No instances of io.micrometer.common.util.internal.logging.LocationAwareSlf4JLogger are allowed in the image heap as this class should be initialized at image runtime. To see how this object got instantiated use --trace-object-instantiation=io.micrometer.common.util.internal.logging.LocationAwareSlf4JLogger.
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.classinitialization.ClassInitializationFeature.checkImageHeapInstance(ClassInitializationFeature.java:140)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.pointsto/com.oracle.graal.pointsto.meta.AnalysisUniverse.replaceObject(AnalysisUniverse.java:583)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.replaceObject(AnalysisConstantReflectionProvider.java:257)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.interceptValue(AnalysisConstantReflectionProvider.java:228)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readValue(AnalysisConstantReflectionProvider.java:105)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantReflectionProvider.readFieldValue(AnalysisConstantReflectionProvider.java:84)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.util.ConstantFoldUtil$1.readValue(ConstantFoldUtil.java:55)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.core.common.spi.JavaConstantFieldProvider.readConstantField(JavaConstantFieldProvider.java:78)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.ameta.AnalysisConstantFieldProvider.readConstantField(AnalysisConstantFieldProvider.java:72)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.nodes.util.ConstantFoldUtil.tryConstantFold(ConstantFoldUtil.java:51)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.ConstantFoldLoadFieldPlugin.tryConstantFold(ConstantFoldLoadFieldPlugin.java:61)
micronaut-projects/micronaut-core#17 32.48       at org.graalvm.nativeimage.builder/com.oracle.svm.hosted.phases.ConstantFoldLoadFieldPlugin.handleLoadStaticField(ConstantFoldLoadFieldPlugin.java:57)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genGetStatic(BytecodeParser.java:4753)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.genGetStatic(BytecodeParser.java:4724)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.processBytecode(BytecodeParser.java:5218)
micronaut-projects/micronaut-core#17 32.48       at jdk.internal.vm.compiler/org.graalvm.compiler.java.BytecodeParser.iterateBytecodesForBlock(BytecodeParser.java:3359)
micronaut-projects/micronaut-core#17 32.48       ... 46 more
micronaut-projects/micronaut-core#17 32.48 ------------------------------------------------------------------------------------------------------------------------
micronaut-projects/micronaut-core#17 32.48                         2.5s (7.8% of total time) in 22 GCs | Peak RSS: 5.40GB | CPU load: 3.24
micronaut-projects/micronaut-core#17 32.48 ========================================================================================================================
micronaut-projects/micronaut-core#17 32.48 Failed generating 'application' after 31.5s.
micronaut-projects/micronaut-core#17 33.28 Error: Image build request failed with exit status 1

```

### Environment Information

- macOS Ventura, Linux
- Micronaut 3.8.0

### Example Application

https://github.com/saw303/micronaut-micrometer-graal-bug

### Version

3.8.0

--- a/gradle.properties
+++ b/gradle.properties
@@ -1,4 +1,4 @@
-projectVersion=4.7.0
+projectVersion=4.7.1-SNAPSHOT
 micronautDocsVersion=2.0.0
 micronautVersion=3.7.3
 micronautTestVersion=3.7.0

--- a/micrometer-core/src/main/resources/META-INF/native-image/io.micronaut.micrometer/micronaut-micrometer-core/native-image.properties
+++ b/micrometer-core/src/main/resources/META-INF/native-image/io.micronaut.micrometer/micronaut-micrometer-core/native-image.properties
@@ -14,5 +14,5 @@
 # limitations under the License.
 #
 
-Args = --initialize-at-build-time=io.micrometer.core \
+Args = --initialize-at-build-time=io.micrometer.core,io.micrometer.common.util.internal.logging.LocationAwareSlf4JLogger \
        --initialize-at-run-time=io.micronaut.configuration.metrics.binder.cache.$MicronautCaffeineCacheMetricsBinder$Definition,io.micronaut.configuration.metrics.binder.cache.$JCacheMetricsBinder$Definition,io.micronaut.configuration.metrics.binder.datasource.$DataSourcePoolMetricsBinderFactory$DataSourceMeterBinder0$Definition

